lib/SILOptimizer/Analysis/ARCAnalysis.cpp

```diff
@@ -1154,3 +1156,132 @@ SILInstruction *swift::findReleaseToMatchUnsafeGuaranteedValue(
   }
   return nullptr;
 }
+
+void EpilogueARCContext::initializeDataflow() {
+  // Find the exit blocks.
+  auto Throw = F->findThrowBB();
+  auto Return = F->findReturnBB();
+  if (Throw != F->end())
+    ExitBlocks.insert(&*Throw);
+  if (Return != F->end())
+    ExitBlocks.insert(&*Return);
+  // Return and throw blocks BBSetOut is initialized to true.
+  for (auto &B : *F) {
+    EpilogueARCBlockStates[&B] =
+              new (BPA.Allocate()) EpilogueARCBlockState();
+  }
+}
+
+void EpilogueARCContext::convergeDataflow() {
+  // Keep iterating until Changed is false.
+  bool Changed = false;
+  do {
+    Changed = false;
+    // Iterate until the data flow converges.
+    for (SILBasicBlock *B : PO->getPostOrder()) {
+      auto BS = EpilogueARCBlockStates[B];
+      // Merge in all the successors.
+      if (!B->succ_empty()) {
+        auto Iter = B->succ_begin();
+        BS->BBSetOut = EpilogueARCBlockStates[*Iter]->BBSetIn;
```

Comment:
you should compute the new out-value in a local variable so that you can compare the new value with the value from the previous iteration. See below